tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -52,50 +101,197 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
           .filter($"b" < 100)
           .sort($"b")
       }
+      val pluginTypeChecker = new PluginTypeChecker()
+      val app = createAppFromEventlog(eventLog)
+      assert(app.sqlPlans.size == 1)
+      app.sqlPlans.foreach { case (sqlID, plan) =>
+        val planInfo = SQLPlanParser.parseSQLPlan(plan, sqlID, pluginTypeChecker, app)
+        assert(planInfo.execInfo.size == 11)
+        val wholeStages = planInfo.execInfo.filter(_.exec.contains("WholeStageCodegen"))
+        assert(wholeStages.size == 6)
+        // only 2 in the above example have projects and filters
+        val numSupported = wholeStages.filter(_.isSupported).size
+        assert(numSupported == 5)
+        assert(wholeStages.forall(_.duration.nonEmpty))
+        val allChildren = wholeStages.flatMap(_.children).flatten
+        assert(allChildren.size == 10)
+        val filters = allChildren.filter(_.exec == "Filter")
+        assertSizeAndSupported(2, filters)
+        val projects = allChildren.filter(_.exec == "Project")
+        assertSizeAndSupported(2, projects)
+        val sorts = allChildren.filter(_.exec == "Sort")
+        assertSizeAndSupported(3, sorts)
+      }
+    }
+  }
+
+  test("FileSourceScan") {
+    val profileLogDir = ToolTestUtils.getTestResourcePath("spark-events-profiling")
```

Comment:
Nit:  Making this class variable would help to reuse in other tests.

Reply:
updated